File: backend/app/models/mission.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.core.database import Base

class Mission(Base):
    __tablename__ = "missions"

    id = Column(Integer, primary_key=True, index=True)
    mission_id = Column(String, unique=True, index=True, nullable=False)
    drone_id = Column(String, index=True)
    status = Column(String, default="PLANNED")  # PLANNED, IN_FLIGHT, COMPLETED, ABORTED
    data_source = Column(String, default="UNKNOWN")  # CSV, ESP32, DEMO
    
    start_time = Column(DateTime, nullable=True)
    end_time = Column(DateTime, nullable=True)
    duration_seconds = Column(Integer, default=0)
    distance_km = Column(Float, default=0.0)
    total_readings = Column(Integer, default=0)
    
    start_latitude = Column(Float, nullable=True)
    start_longitude = Column(Float, nullable=True)
    current_latitude = Column(Float, nullable=True)
    current_longitude = Column(Float, nullable=True)
    
    created_at = Column(DateTime, default=datetime.utcnow)

    readings = relationship("Reading", back_populates="mission", cascade="all, delete-orphan")
    hotspots = relationship("Hotspot", back_populates="mission", cascade="all, delete-orphan")

File: backend/models/Mission.js
const mongoose = require('mongoose');

const missionSchema = new mongoose.Schema({
  mission_id: { type: String, required: true, unique: true, index: true },
  drone_id: { type: String, index: true },
  status: { type: String, default: "PLANNED" }, // PLANNED, IN_FLIGHT, COMPLETED, ABORTED
  data_source: { type: String, default: "UNKNOWN" }, // CSV, ESP32, DEMO
  
  start_time: { type: Date, default: null },
  end_time: { type: Date, default: null },
  duration_seconds: { type: Number, default: 0 },
  distance_km: { type: Number, default: 0.0 },
  total_readings: { type: Number, default: 0 },
  
  start_latitude: { type: Number, default: null },
  start_longitude: { type: Number, default: null },
  current_latitude: { type: Number, default: null },
  current_longitude: { type: Number, default: null },
  
  created_at: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Mission', missionSchema);
